docs/framework/wcf/feature-details/data-contract-schema-reference.md

```diff
@@ -1,680 +1,691 @@
+
+### \<xs:simpleType>: attributes
+
+|Attribute|Schema|
+|---------------|------------|
+|`final`|Ignored.|
+|`id`|Ignored.|
+|`name`|Supported, maps to the data contract name.|
+
+### \<xs:simpleType>: contents
+
+|Contents|Schema|
+|--------------|------------|
+|`restriction`|Supported. Maps to enumeration data contracts. This attribute is ignored if it does not match the enumeration pattern. See the `xs:simpleType` restrictions section.|
+|`list`|Supported. Maps to flag enumeration data contracts. See the `xs:simpleType` lists section.|
+|`union`|Forbidden.|
+
+### \<xs:restriction>
+
+- Complex type restrictions are supported only for base="`xs:anyType`".
+
+- Simple type restrictions of `xs:string` that do not have any restriction facets other than `xs:enumeration` are mapped to enumeration data contracts.
+
+- All other simple type restrictions are mapped to the types they restrict. For example, a restriction of `xs:int` maps to an integer, just as `xs:int` itself does. For more information about primitive type mapping, see Type/primitive mapping.
+
+### \<xs:restriction>: attributes
+
+|Attribute|Schema|
+|---------------|------------|
+|`base`|Must be a supported simple type or `xs:anyType`.|
+|`id`|Ignored.|
+
+### \<xs:restriction> for all other cases: contents
+
+|Contents|Schema|
+|--------------|------------|
+|`simpleType`|If present, must be derived from a supported primitive type.|
+|`minExclusive`|Ignored.|
+|`minInclusive`|Ignored.|
+|`maxExclusive`|Ignored.|
+|`maxInclusive`|Ignored.|
+|`totalDigits`|Ignored.|
+|`fractionDigits`|Ignored.|
+|`length`|Ignored.|
+|`minLength`|Ignored.|
+|`maxLength`|Ignored.|
+|`enumeration`|Ignored.|
+|`whiteSpace`|Ignored.|
+|`pattern`|Ignored.|
+|(blank)|Supported.|
+
+## Enumeration
+
+### \<xs:restriction> for enumerations: attributes
+
+|Attribute|Schema|
+|---------------|------------|
+|`base`|If present, must be `xs:string`.|
+|`id`|Ignored.|
+
+### \<xs:restriction> for enumerations: contents
+
+|Contents|Schema|
+|--------------|------------|
+|`simpleType`|If present, must be an enumeration restriction supported by the data contract (this section).|
+|`minExclusive`|Ignored.|
+|`minInclusive`|Ignored.|
+|`maxExclusive`|Ignored.|
+|`maxInclusive`|Ignored.|
+|`totalDigits`|Ignored.|
+|`fractionDigits`|Ignored.|
+|`length`|Forbidden.|
+|`minLength`|Forbidden.|
+|`maxLength`|Forbidden.|
+|`enumeration`|Supported. Enumeration "id" is ignored, and "value" maps to the value name in the enumeration data contract.|
+|`whiteSpace`|Forbidden.|
+|`pattern`|Forbidden.|
+|(empty)|Supported, maps to empty enumeration type.|
+
+ The following code shows a C# enumeration class.
+
+```csharp
+public enum MyEnum
+{
+  first = 3,
+  second = 4,
+  third =5
+}
+```
+
+This class maps to the following schema by the `DataContractSerializer`. If enumeration values start from 1, `xs:annotation` blocks are not generated.
+
+```xml
+<xs:simpleType name="MyEnum">
+<xs:restriction base="xs:string">
```

Comment:
Levels of indentation here should be fixed. \<xs:restriction> is a child of \<xs:simpletype>.

Reply:
OK, I left that as-is, but i'll go fix that now

Reply:
Ran an XML formatter on this block. Left the other ones since it made it a little noisy by pulling attributes to the same line